delta/plugins/graph-analytics/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/graph/analytics/indexing/GraphAnalyticsStream.scala

```diff
@@ -0,0 +1,106 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.graph.analytics.indexing
+
+import cats.data.NonEmptyList
+import cats.syntax.all._
+import ch.epfl.bluebrain.nexus.delta.kernel.database.Transactors
+import ch.epfl.bluebrain.nexus.delta.plugins.graph.analytics.indexing.GraphAnalyticsResult.{Index, Noop, UpdateByQuery}
+import ch.epfl.bluebrain.nexus.delta.plugins.graph.analytics.model.JsonLdDocument
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.Files
+import ch.epfl.bluebrain.nexus.delta.plugins.storage.files.model.FileState
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sdk.resources.Resources
+import ch.epfl.bluebrain.nexus.delta.sdk.resources.model.ResourceState
+import ch.epfl.bluebrain.nexus.delta.sourcing.config.QueryConfig
+import ch.epfl.bluebrain.nexus.delta.sourcing.implicits._
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.{ElemStream, EntityType, ProjectRef, Tag}
+import ch.epfl.bluebrain.nexus.delta.sourcing.offset.Offset
+import ch.epfl.bluebrain.nexus.delta.sourcing.query.StreamingQuery
+import doobie._
+import doobie.implicits._
+import io.circe.Json
+import monix.bio.{Task, UIO}
+
+trait GraphAnalyticsStream {
+
+  /**
+    * Allows to generate a non-terminating [[GraphAnalyticsResult]] stream for the given project
+    * @param project
+    *   the project to stream from
+    * @param start
+    *   the offset to start with
+    */
+  def apply(project: ProjectRef, start: Offset): ElemStream[GraphAnalyticsResult]
+
+}
+
+object GraphAnalyticsStream {
+
+  private val emptyMapType = Map.empty[Iri, Set[Iri]]
+
+  /**
+    * We look for a resource with these ids in this project and return their types if it can be found
+    */
+  private def query(project: ProjectRef, xas: Transactors)(nel: NonEmptyList[Iri]): UIO[Map[Iri, Set[Iri]]] = {
+    val inIds = Fragments.in(fr"id", nel)
+    sql"""
+         | SELECT id, value->'types'
+         | FROM public.scoped_states
+         | WHERE org = ${project.organization}
+         | AND project = ${project.project}
+         | AND $inIds
+         | AND tag = ${Tag.latest.value}
+         |""".stripMargin
+      .query[(Iri, Json)]
+      .to[List]
+      .transact(xas.streaming)
+      .map { l =>
+        l.foldLeft(emptyMapType) { case (acc, (id, json)) =>
+          acc ++ json.as[Set[Iri]].toOption.map(id -> _)
+        }
+      }
+  }.hideErrors
+
+  /**
+    * We batch ids and looks for existing resources
+    */
+  private[indexing] def findRelationships(project: ProjectRef, xas: Transactors, batchSize: Int)(
+      ids: Set[Iri]
+  ): UIO[Map[Iri, Set[Iri]]] = {
+    val groupIds = NonEmptyList.fromList(ids.toList).map(_.grouped(batchSize).toList)
+    val noop     = UIO.pure(emptyMapType)
+    groupIds.fold(noop) { list =>
+      list.foldLeftM(emptyMapType) { case (acc, l) =>
+        query(project, xas)(l).map(_ ++ acc)
+      }
+    }
+  }
+
+  // $COVERAGE-OFF$
```

Comment:
I don't test that whole part as a unit test here:
* The subparts are already tested independently
* It would need to set up the resources and the files, so the storages and so on for little value
This will be tested through integration tests